embedded-graphics-core/src/geometry/mod.rs

```diff
@@ -0,0 +1,36 @@
+//! Geometry module.
+
+mod point;
+mod size;
+
+pub use point::Point;
+pub use size::Size;
+
+use crate::rectangle::Rectangle;
+
+/// Adds the ability to get the dimensions/position of a graphics object
+///
+/// This **should** be implemented for all builtin embedded-graphics primitives and fonts. Third party
+/// implementations do not have to implement this trait as an object may not have a known size. If
+/// the object _does_ have a known size, this trait **should** be implemented.
```

Comment:
This needs to be updated. `Dimensions` is also used by `DrawTarget`, where this description doesn't make much sense.

IMO this is one of the places were we need to consider the different target audiences for the docs. One the one hand we have normal users who need to known how to use `Dimensions` to get the dimensions of drawable and draw targets. On the other hand we have authors of third party crates who need to know when and how the need to implement `Dimensions` and `OriginDimensions`. I think this is an important distinction and should also be considered for similar traits. I would suggest to put the user facing docs first and than add a common header like `# Implementing Dimensions` or `# Implementation notes` followed by the third party crate developer docs. I've started to follow this pattern in the docs for `ImageDrawable` (maybe not the greatest example).

Reply:
I like the `implementation notes` idea, although I could use some suggestions for what goes above that heading, and below it.

Reply:
Above:

* What objects the can this trait be used on.
* A description of which dimension is defined by this trait: For styled primitives it's the bounding box around all drawn pixels, but for draw target it's the drawable area.
* The difference between the dimensions of styled and unstyled primitives.
* Maybe some suggestions how the result can be used, like being able to use `.bounding_box().points()`.
* At some point we might also want to add examples to this, but that's even more out of scope of this PR than the rest of this suggestion :wink:.

Below:

* That some traits require this trait to be implemented, perhaps with a list of other traits.
* Your description of `OriginDimensions` isn't accurate: The `top_left` is also known and is guaranteed to be `Point::zero()`.
* Maybe a suggestion that the impl of `Dimensions` for drawables can be tested by using `MockDisplay::affeted_area`.